pkg/resourcegenerator/core/pod.go

```diff
@@ -212,20 +217,19 @@ func getContainerPorts(application *skiperatorv1alpha1.Application, opts PodOpts
 	return containerPorts
 }
 
-func affinityTermForAppAndKey(appName string, key SkiperatorTopologyKey) corev1.WeightedPodAffinityTerm {
-	return corev1.WeightedPodAffinityTerm{
-		Weight: 100,
-		PodAffinityTerm: corev1.PodAffinityTerm{
-			LabelSelector: &v1.LabelSelector{
-				MatchExpressions: []v1.LabelSelectorRequirement{
-					{
-						Key:      "app",
-						Operator: v1.LabelSelectorOpIn,
-						Values:   []string{appName},
-					},
+func spreadConstraintForAppAndKey(appName string, key SkiperatorTopologyKey) corev1.TopologySpreadConstraint {
+	return corev1.TopologySpreadConstraint{
+		MaxSkew:           1,
+		TopologyKey:       string(key),
+		WhenUnsatisfiable: corev1.ScheduleAnyway,
+		LabelSelector: &v1.LabelSelector{
+			MatchExpressions: []v1.LabelSelectorRequirement{
+				{
+					Key:      "app",
+					Operator: v1.LabelSelectorOpIn,
+					Values:   []string{appName},
 				},
```

Comment:
The short answer is no.

Since these topology spread constraints are applied to a `spec`, it is only applied in the namespace specified by the outer resource (`StatefulSet`, `Deployment` etc.)

Reply:
Ah! 💡 👍